src/MSONValueMemberParser.cc

```diff
@@ -88,7 +88,9 @@ namespace snowcrash {
             IntermediateParseResult<mson::Mixin> mixin(sections.report);
             cur = MSONMixinParser::parse(node, siblings, pd, mixin);
 
-            if (baseType != mixin.node.baseType) {
+            if (!((baseType == mson::ImplicitObjectBaseType && mixin.node.baseType == mson::ObjectBaseType) ||
```

Comment:
Can you make it into a inline function so that it is easier to see what it does? Like `IsSameType(baseType, mixin.node.baseType)` where you put all the conditions. It also could be then reused if needed otherwise we might bump into this once again.